core/src/main/java/org/apache/calcite/rel/rules/AggregateJoinTransposeRule.java

```diff
@@ -336,8 +346,9 @@ public void onMatch(RelOptRuleCall call) {
     relBuilder.project(projects);
 
     boolean aggConvertedToProjects = false;
-    if (allColumnsInAggregate) {
+    if (allColumnsInAggregate && join.getJoinType() != JoinRelType.FULL) {
```

Comment:
`&& join.getJoinType() != JoinRelType.FULL) {` is not needed anymore if we bail out correctly above?

Reply:
@jcamachor Correct, I'll remove this.